src/javascript/campaign-url-builder/components/problematic.js

```diff
@@ -0,0 +1,162 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This file manages "problematic" URLs. For instance, if users are
+ * creating links to Play Store URLs, they should user the Google Play
+ * Url Builder (https://goo.gl/qycyHd) instead of the GA devtools app.
+ *
+ * See b/69681865 for more details.
+ */
+
+import React from 'react';
+import {isRegExp} from 'lodash';
+
+
+/**
+ * Create a function that, given a RegExp instance, returns a function
+ * that tests a url against that regex.
+ *
+ * @param  {RegExp} regexp compiled regular expression object used in
+ * the test
+ * @return {Function} function that tests the given URL
+ * against the regex
+ */
+const matchesRegexp = regexp => url => regexp.test(url);
+
+/**
+ * Given either a RegExp or a pattern, return a function that tests if
+ * a given URL matches that pattern.
+ *
+ * @param  {string|Regexp} pattern to use in the returned testing
+ * function. If the pattern is a string, it is compiled (case-insensitive)
+ * to a Regexp
+ * @return {Function} function that tests if a URL matches
+ * the given regexp
+ */
+const matches = pattern => isRegExp(pattern) ?
+  matchesRegexp(pattern) :
+  matchesRegexp(new RegExp(pattern, 'i'));
+
+
+/**
+ * Given a domain, create a function that returns true if the domain
+ * matches a given URL. The domain is matched exactly, and can optionally
+ * be prepended with http:// or https://.
+ *
+ * @param {string} domain the domain to match. Can be a regex pattern, but all
+ * . characters are unconditionally escaped
+ *
+ * @return {Function} a function that returns true if the
+ * domain matches the given url exactly
+ */
+const domainMatches = domain =>
+  matches('^(?:https?://)?(:?' + domain.replace('.', '\\.') + ')(?:$|[/?#])');
+
+/**
+ * Given a domain suffix, create a function that returns true if the
+ * domain matches a given URL. The domain is matched as a suffix; that
+ * is, the domain "google.com" will match "play.google.com",
+ * "mail.google.com", etc, as well as just "google.com".
+ *
+ * @param {string} suffix Suffix of the domain to match. Can be a pattern,
+ * but all . characters are unconditionally escaped
+ *
+ * @return {Function} a function that returns true if the
+ * domain matches the given url as a suffix
+ */
+const domainSuffix = suffix =>
+  domainMatches(`(?:[a-z0-9-]+.)*(?:${suffix})`);
+
+const googlePlayBuilderUrl = 'https://developers.google.com/analytics/devguides/collection/android/v4/campaigns#google-play-url-builder';
```

Comment:
The URL lines? I agree, but I couldn't think of a way to make them shorter. This seemed to me like a reasonable exception to the line length rule, rather than trying to break up the URLs into multiple lines.

Reply:
I can only assume that eslint has an exception to the line length rule when the whole line is assigning a very long string to a variable.

Reply:
I think the style guide says to just use string concatenation. I know it makes an exception for URLs in comments, but I don't think it makes an exception for URLs in strings (though I'm not 100% sure about that).